File: Robert Martin (Uncle Bob)/Clean Architecture p117 (Abstract Factory).cs
// Interfaz Service
public interface IService
{
    // Métodos de Service
}

// Clase ConcreteImpl que implementa IService
public class ConcreteImpl : IService
{
    // Implementación de ConcreteImpl
}

// Interfaz ServiceFactory
public interface IServiceFactory<I>
{
    I MakeSvc();
}

// Clase ServiceFactoryImpl que implementa IServiceFactory<IService>
public class ServiceFactoryImpl : IServiceFactory<IService>
{
    public IService MakeSvc()
    {
        return new ConcreteImpl();
    }
}

// Clase Application
public class Application
{
    private readonly IServiceFactory<IService> serviceFactory;

    public Application(IServiceFactory<IService> serviceFactory)
    {
        this.serviceFactory = serviceFactory;
    }

    // Métodos y lógica de Application
}

File: [ BATTAGLIA ]/Clean Architecture/Clean Architecture p117 (Abstract Factory).java
// Interfaz Service
public interface Service {
    // Métodos de Service
}

// Clase ConcreteImpl que implementa Service
public class ConcreteImpl implements Service {
    // Implementación de ConcreteImpl
}

// Interfaz ServiceFactory
public interface ServiceFactory<I> {
    I makeSvc();
}

// Clase ServiceFactoryImpl que implementa ServiceFactory<Service>
public class ServiceFactoryImpl implements ServiceFactory<Service> {
    public Service makeSvc() {
        return new ConcreteImpl();
    }
}

// Clase Application
public class Application {
    private final ServiceFactory<Service> serviceFactory;

    public Application(ServiceFactory<Service> serviceFactory) {
        this.serviceFactory = serviceFactory;
    }

    // Métodos y lógica de Application
}
